packages/performance/src/definitions.ts

```diff
@@ -105,3 +148,182 @@ export interface IsEnabledResult {
    */
   enabled: boolean;
 }
+
+/**
+ * @since 6.3.0
+ */
+export interface PutAttributeOptions {
+  /**
+   * Name of the trace to set its attribute.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * Name of the attribute to set its value.
+   *
+   * @since 6.3.0
+   * @example "experiment"
+   */
+  attribute: string;
+  /**
+   * The value to set to the attribute.
+   *
+   * @since 6.3.0
+   * @example "A"
+   */
+  value: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributeOptions {
+  /**
+   * Name of the trace to set its attribute.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * Name of the attribute to retrieve its value.
+   *
+   * @since 6.3.0
+   */
+  attribute: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributeResult {
+  /**
+   * The value of the custom attribute.
+   *
+   * @since 6.3.0
+   */
+  value: string | null;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributesOptions {
+  /**
+   * Name of the trace to get its attributes.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetAttributesResult {
+  /**
+   * A map of all custom attributes of a trace with their values.
+   *
+   * @since 6.3.0
+   */
+  result: { [key: string]: string };
+}
+
+/**
+ * @since 6.3.0
+ */
+export type RemoveAttributeOptions = GetAttributeOptions;
+
+/**
+ * @since 6.3.0
+ */
+export interface PutMetricOptions {
+  /**
+   * Name of the trace to set its metric.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * The metric name.
+   *
+   * @since 6.3.0
+   */
+  metricName: string;
+  /**
+   * The value to set for the metric.
+   * The given value is floored down to the nearest integer.
+   *
+   * @since 6.3.0
+   */
+  num: number;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetMetricOptions {
+  /**
+   * Name of the trace to get its metric.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * The metric name.
+   *
+   * @since 6.3.0
+   */
+  metricName: string;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface GetMetricResult {
+  /**
+   * The value of the metric if exists.
+   *
+   * @since 6.3.0
+   */
+  value: number;
+}
+
+/**
+ * @since 6.3.0
+ */
+export interface RecordOptions {
+  /**
+   * Name of the trace to record.
+   *
+   * @since 6.3.0
+   */
+  traceName: string;
+  /**
+   * Start time of the trace since epoch in milliseconds.
+   *
+   * @since 6.3.0
+   */
+  startTime: number;
+  /**
+   * The duration of the trace in milliseconds.
+   *
+   * @since 6.3.0
+   */
+  duration: number;
+  /**
+   * An optional object that holds optional maps of custom metrics and attributes.
+   *
+   * @since 6.3.0
+   */
+  options?: {
+    /**
+     * An optional map of metrics to be set on the trace.
+     */
+    metrics?: { [key: string]: number };
+    /**
+     * An optional map of attributes to be set on the trace.
```

Comment:
```suggestion
     * An optional map of attributes to be set on the trace.
     *
     * @since 6.3.0
```

Please always set the `@since` tag.